tests/research/test_run_repository.py

```
@@ -120,6 +127,304 @@ def test_model_diagnostics_parser_preserves_direction_diagnostics():
+    )
+
+    assert partition_sizes == [500]
+    assert row_batch_sizes == [500, 500, 1]
+    assert len(statements) == 4
+    assert [snapshot["run_id"] for snapshot in snapshots] == [
+        f"run-{index:04d}" for index in range(1001)
+    ]
+
+
+def test_project_persisted_snapshot_requires_private_metadata_keys():
+    with pytest.raises(ValueError, match="_artifact_presence.*_version_pack_values"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)